packages/frontend/src/components/dialogs/ProxyConfiguration/index.tsx

```diff
@@ -0,0 +1,403 @@
+        setConnectivityStatus(connectivity)
+        removeConnectivityListener = onDCEvent(
+          accountId,
+          'ConnectivityChanged',
+          () =>
+            debounceWithInit(async () => {
+              const connectivity =
+                await BackendRemote.rpc.getConnectivity(accountId)
+              setConnectivityStatus(connectivity)
+            }, 300)()
+        )
+      }
+    }
+    checkConnectivity()
+    return () => {
+      removeConnectivityListener()
+    }
+  }, [accountId, configured])
+
+  /**
+   * Update proxy settings in the backend
+   * called after any proxyState change
+   */
+  useEffect(() => {
+    if (!proxyState.updateSettings) {
+      // don't update while loading values from backend
+      return
+    }
+    const updateProxySettings = async () => {
+      const proxyString =
+        proxyState.proxies.length > 0
+          ? [
+              proxyState.activeProxy,
+              ...proxyState.proxies.filter(p => p !== proxyState.activeProxy),
+            ]
+              .filter(s => !!s) // remove null & empty strings
+              .join('\n')
+          : ''
+      if (proxyState.enabled && proxyString.trim() === '') {
+        openAlertDialog({
+          message: tx('proxy_invalid'),
+        })
+        return
+      }
+      try {
+        await BackendRemote.rpc.batchSetConfig(accountId, {
+          proxy_url: proxyString,
+          proxy_enabled: proxyState.enabled ? Proxy.ENABLED : Proxy.DISABLED,
+        })
+
+        await BackendRemote.rpc.stopIo(accountId)
+        await BackendRemote.rpc.startIo(accountId)
+      } catch (error) {
+        log.error('failed to update proxy settings', error)
+        openAlertDialog({
+          message: unknownErrorToString(error),
+        })
+      }
+    }
+    updateProxySettings()
+  }, [
+    proxyState.enabled,
+    accountId,
+    openAlertDialog,
+    tx,
+    proxyState.proxies,
+    proxyState.activeProxy,
+    proxyState.updateSettings,
+    proxyState,
+  ])
+
+  /**
+   * validate settings before closing
+   * to avoid invalid settings
+   */
+  const closeDialog = () => {
+    if (proxyState.enabled && proxyState.proxies.length === 0) {
+      openAlertDialog({
+        message: tx('proxy_invalid'),
+      })
+      return
+    }
+    onClose()
+  }
+
+  const copyToClipboard = useCallback(
+    (url: string) => {
+      navigator.clipboard.writeText(url)
+      openAlertDialog({
+        message: `${url}\n${tx('copied_to_clipboard')}`,
+      })
+    },
+    [tx, openAlertDialog]
+  )
+
+  // some basic validations, returns true if the url seems valid
+  // and is not already in the list of proxies
+  const maybeValidProxyUrl = (url: string): boolean => {
+    const parts = url.split('://')
+    return (
+      parts.length === 2 &&
+      parts[0].length >= 2 && // shortest protocol is ss://
+      parts[1].length >= 1 && // host
+      !proxyState.proxies.includes(url)
+    )
+  }
```

Comment:
IDK about adding this frontend validation. This should be a job for the back end. What if we add support for a format that does not conform to this pattern?